### PR TITLE
Support matching against wildcard characters in acls urls.

### PR DESCRIPTION
**Pull request by dkschoonover**
### Summary
Allow wildcard characters to be specified in acls urls.  This can be used to match against any domain regardless of subdomain.  Any number of valid domain characters can be matched against the asterisk.

### Additional Details
I tried to keep the regex as simple as possible.  Definitely let me know if stronger validation is needed.

### CODE REVIEW

File: src/provider/application.js [hunk 1]
```diff
@@ -215,7 +215,10 @@ class Application extends EventEmitter {
       this.activeACL = origin;
     }
 
-    if (this.acls.includes('*') || this.acls.includes(origin)) {
+    if (this.acls.includes('*') || this.acls.includes(origin) || this.acls.some((acl) => {
+      const aclRegExp = new RegExp(acl.replace('*', '[a-zA-Z0-9-._]*'));
```
Comment (mhemesath) [inline src/provider/application.js:219]: This seems to fail if i do the following scenario
```js
const acl = "*.foo.com"
const aclRegExp = new RegExp(acl.replace('*', '[a-zA-Z0-9-._]*'));
"bar.foo.com.evil.com".match(aclRegExp);
```

I'm wondering if a safer approach is the following.
```
const subdomain = acl.replace(/^\*\/, ''); // Strip leading wildcard to get subdomain
return origin.endsWith(subdomain); // Verify origin is subdomain. Method needs polyfil for IE
```

Comment (dkschoonover) [reply src/provider/application.js]: I like this idea.  I have made the changes and updated the documentation to specify the wildcard character is only supported as the first character of the string.

File: src/provider/application.js [hunk 2]
```diff
@@ -215,7 +215,11 @@ class Application extends EventEmitter {
       this.activeACL = origin;
     }
 
-    if (this.acls.includes('*') || this.acls.includes(origin)) {
+    if (this.acls.includes('*') || this.acls.includes(origin) || this.acls.some((acl) => {
+      // Strip leading wildcard to get domain and verify it matches the end of the event's origin.
+      const domain = acl.replace(/^\*/, '');
+      return origin.substring(origin.length - domain.length) === domain;
```
Comment (dkschoonover) [inline src/provider/application.js:221]: If origin is shorter than domain, the entire origin is returned from substring.

**Comment by mhemesath** [comment]
@foxannefoxanne  can you cut a release for this as a minor bump?

**Comment by dkschoonover** [comment]
Thanks to all for helping get these changes in!

**Review by foxannefoxanne — APPROVED** [review]

**Pull request merged**